### PR TITLE
Separate flow test streamlit from executable

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
1. Fix when connection.provider="azureml"
2. Separate flow test streamlit from executable

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2 [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import json
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2:1]: maybe add some comments on purpose of these different files

File: src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2 [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    st.title("{{flow_name}}")
+    st.divider()
+    st.chat_message("assistant").write("Hello, please input following flow inputs.")
+    container = st.container()
+    with container:
+        show_conversation()
+
+    with st.form(key='input_form', clear_on_submit=True):
+        st.text('{{chat_input_name}}')
+        {{chat_input_name}} = st_quill(html=True, toolbar=["image"], key='{{chat_input_name}}', placeholder='Please enter the list values and use the image icon to upload a picture. Make sure to format each list item correctly with line breaks')
```
Comment (YingChen1996) [inline src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2:90]: Are all inputs list formats in chat_flow?

Comment (brynn-code) [reply src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2:90]: Yes, there is only one input is allowed in chat flow, and also one output.

Comment (wangchao1230) [reply src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2:90]: and the one input should always be type list?

Comment (wangchao1230) [reply src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2:91]: where do we do the check?

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2 (hunk 2 above)
Comment (brynn-code) [reply src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2:91]: At StreamlitFileGenerator(BaseGenerator) init, _is_chat_flow()

**Review by lalala123123 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**